File: ExamFront/src/app/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  constructor(private _http:HttpClient) { }

  public categories()
  {
    return this._http.get(`${baseUrl}/category/`)
  }

  public addCategory(category:any)
  {
    return this._http.post(`${baseUrl}/category/`,category)
  }

  public updateCategory(category:any)
  {
    return this._http.put(`${baseUrl}/category/`,category);
  }
  public getCategoryById(cid: number) {
    return this._http.get(`${baseUrl}/category/${cid}`);
  }

  public deleteCategory(categoryId:any)
  {
    return this._http.delete(`${baseUrl}/category/${categoryId}`);
  }
}

File: examServer/src/main/java/com/prashant/exam/examServer/Service/CategoryService.java
package com.prashant.exam.examServer.Service;

import java.util.Set;

import com.prashant.exam.examServer.Entity.exam.Category;

public interface CategoryService {
	
	public Category addCategory(Category category);
	
	public Category updateCategory(Category category);
	
	public Set<Category> getCategories();
	
	public Category getCategory(Long categoryId);
	
	public void deleteCategory(Long categoryId);

}
